components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -0,0 +1,613 @@
+        }
+        try {
+            RelayUtils.buildMessage(axis2MC);
+            if (logger.isDebugEnabled()) {
+                logger.debug("Successfully built the request message");
+            }
+            if (!ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)) {
+                return true;
+            }
+            try {
+                initialize(messageContext);
+            } catch (APIManagementException e) {
+                logger.error("Error occurred while initializing the swagger elements", e);
+            }
+            if (swagger != null) {
+                requestMethod = messageContext.getProperty(APIMgtGatewayConstants.
+                        ELECTED_REQUEST_METHOD).toString();
+                if (!APIConstants.SupportedHTTPVerbs.GET.name().equals(requestMethod)) {
+                    try {
+                        validateRequest(messageContext);
+                    } catch (APIManagementException e) {
+                        logger.error("Error occurred while validating the API request", e);
+                    }
+                }
+            }
+        } catch (IOException | XMLStreamException e) {
+            logger.error("Error occurred while building the API request", e);
+        }
+        return true;
+    }
+
+    /**
+     * Validate the response message.
+     *
+     * @param messageContext Context of the API response
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleResponse(MessageContext messageContext) {
+        logger.info("Validating the API response  Body content..");
+        org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                getAxis2MessageContext();
+        try {
+            RelayUtils.buildMessage(axis2MC);
+            logger.info("Successfully built the response message");
+
+        } catch (IOException e) {
+            logger.error("Error occurred while building the API response", e);
+        } catch (XMLStreamException e) {
+            logger.error("Error occurred while validating the API response", e);
+        }
+        Object objectResponse = axis2MC.getProperty(ThreatProtectorConstants.REST_CONTENT_TYPE);
+        if (objectResponse == null) {
+            return true;
+        }
+        String contentType = objectResponse.toString();
+        if (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)) {
+            try {
+                validateResponse(messageContext);
+            } catch (APIManagementException e) {
+                logger.error("Error occurred while validating the API response", e);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Initialize the swagger from the Local Entry.
+     *
+     * @param messageContext Content of the message context
+     */
+    private void initialize(MessageContext messageContext) throws APIManagementException {
```

Comment:
i think this is already done by the JWT feature.

Reply:
fixed.